lib/publiccloud/instance.pm

```diff
@@ -209,16 +209,20 @@ sub wait_for_ssh
     $args{timeout}            //= 600;
     $args{proceed_on_failure} //= 0;
     $args{username}           //= $self->username();
+    $args{migration}          //= 0;
     my $start_time = time();
 
     # Check port 22
-    while ((my $duration = time() - $start_time) < $args{timeout}) {
+    while ((my $duration = time() - $start_time) < ($args{timeout} + $args{migration})) {
         last if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0);
         sleep 1;
     }
 
     # Check ssh command
-    while ((my $duration = time() - $start_time) < $args{timeout}) {
+    while ((my $duration = time() - $start_time) < ($args{timeout} + $args{migration})) {
+        if ($args{migration}) {
+            sleep $args{migration};
```

Comment:
same question here , I would like to see failures which cause this . also I would not add second sleep but simply increase sleep below